syntheticEof.ts

```diff
@@ -0,0 +1,83 @@
+import { Transform } from 'node:stream';
+
+// keep in sync with lib.rs::SYNTHETIC_EOF
+export const SYNTHETIC_EOF = Buffer.from('\x1B]7878\x1B\\');
+
+function getCommonPrefixLength(buffer: Buffer) {
+  for (let prefixLen = SYNTHETIC_EOF.length; prefixLen >= 1; prefixLen--) {
+    const suffix = buffer.subarray(buffer.length - prefixLen);
+    const prefix = SYNTHETIC_EOF.subarray(0, prefixLen);
+
+    if (suffix.equals(prefix)) {
+      return prefixLen;
+    }
+  }
```

Comment:
If you wanna optimize this we can probably loop over the max of length of the buffer or `SYNTHETIC_EOF`

Reply:
then check byte by byte (isntead of creating new buffers and calling equals)